### DISCUSSION TITLE
Vtherm compatible boitier Tuya Konyks Ecosy ?

### DISCUSSION THREAD

**Discussion by SchottyOne** [jmcollin78/versatile_thermostat#209] [Q&A]
Hello, avant d'ouvrir un ticket je tente ma chance ici.
Merci pour ce thermostat qui parait correspondre à ce que je cherchais mais voila, il semble que cela ne soit pas compatible avec le boitier qui gère mes convecteurs.
Il se trouve que je dispose de boitiers[ Tuya Konyks Ecosy](https://konyks.com/produit/ecosy/) pour piloter mes convecteurs avec fil pilote. 
Ces derniers permettent notamment de contrôler l'ensemble des modes du convecteur (On/Off, Confort, Eco, Hors gel, -1 ou -2°C) mais la température de consigne ne peut être réglée que sur le convecteur lui-même.
L'intégration sous HA est faite via le module Tuya Officiel (Local Tuya n'étant pas full compatible avec les modes du boitier) Et permet de reprendre le contrôle des boitiers et donc des modes.
Je passais par l'intégration "scheduler" pour gérer les changement de modes mais le hic est qu'avec tout cela, il n'y pas de notion de thermostat et donc de contrôle plus fin de la façon dont les convecteurs fonctionnent. Par exemple, il arrive que je passe en mode confort en fonction du planning et que le convecteur va se mettre à chauffer malgré que la température de la pièce soit déjà supérieure à la consigne (le convecteur semble même ne pas réagir instinctivement en fonction de sa propre consigne..)
Je suis tombé sur Vtherm qui parait correspondre à ce je cherche, à savoir gérer les modes en fonction des relevés de température autre que celui du convecteur (sonde température pour chaque pièce et météo pour la température extérieure) Ainsi que les ouvertures de fenêtre et la présence.
Tout cela est nickel mais visiblement Vtherm ne semble pas pouvoir piloter les boitiers de façon autonome en fonction des consignes et des modes pré réglés.
Le mode manuel fonctionne en revanche via la card [Versatile Thermostat UI Card](https://github.com/jmcollin78/versatile-thermostat-ui-card). Je peux lancer le on/off et les divers modes sans soucis donc il y a quand même une prise en compte des mes entités des boitiers.
Il semble que mon souci provienne du fait que je suis à cheval entre un mode "over_switch" et "over_climate"
En effet, le boitiers ne gérant que les modes du convecteurs on pourrait considérer qu'ils ont plus une fonction de switch. 
Simplement, ils sont intégrés sous HA en tant qu'entité "climate." 
Du coup, j'ai cru comprendre que Vtherm "over climate" ne gérait pas réellement le sous-jacent mais lui apportait plutôt des fonctions supplémentaires tel que la détection d'ouverture.
Effectivement à l'ouverture d'une fenêtre, la commande est bien envoyée au convecteur pour son arrêt.
J'ai cru comprendre également qu'en utilisant le mode de régulation, il serait possible de piloter le sous-jacent mais j'imagine que cela ne lui fournit qu'une nouvelle température de consigne.
De ce fait, je suis bloqué pour piloter les convecteurs via les modes pré réglés car aucun effet sur les boitiers.

En fonction de vos retours d'expérience, y aurait-il une solution pour résoudre cette incompatibilité ? 

- passer par du custom pour transformer mes entités climate pour les passer en mode switch
- Modifier la conf du VTherm pour permettre de gérer les entités climate via le mode over_switch
- utiliser Vtherm via le scheduler pour piloter indirectement les boitiers 

Je suis un peu perdu à trouver la bonne alternative et je suis preneur de toute idée. Désolé pour la longueur de ce message et merci à toutes les âmes charitables.

**Comment by jmcollin78** [comment]
Hello @SchottyOne ,

Je vois que tu as bien creusé le sujet avant de venir ici et c'est une bonne chose. Ca va permettre de se comprendre beaucoup plus facilement.

Ce que font les gens qui ont des radiateurs avec un fil pilote et un boitier qui le pilote à travers le fil pilote c'est d'utiliser le VTherm (Versatile Thermostat) en mode over_switch qui va allumer / éteindre le radiateur en changeant de mode. Je m'explique :
1. on n'utilise que 2 modes: Eteint (Off)  et Comfort que tu règles à 24°,
2. le VTherm va alors demander au boitier de switcher entre ces 2 modes : si il est fait froid, il met plus de Comfort et plus il fait chaud plus il baisse le pourcentage de temps à Comfort. C'est basé sur un cycle, et à chaque cycle, on allume un certain pourcentage du temps le radiateur.

Pour faire ça il faut que tu ais accès à un switch qui va commander ton boitier. Beaucoup de boitier ont ça naturellement mais si ce n'est pas le cas tu vas trouver ici : https://forum.hacf.fr/t/nouveau-thermostat-type-proportionnel-avec-gestion-des-presets-portes-et-fenetres-detection-de-mouvement-gestion-de-presence-et-surconsommation/19160/643

Ca doit ressembler à ça:
```
- platform: template
  switches:
    radiateur_soan:
        friendly_name: radiateur_soan_inv
        value_template: "{{ is_state('switch.radiateur_soan', 'off') }}"
        turn_on:
          service: switch.turn_off
          data:
            entity_id: switch.radiateur_soan
        turn_off:
          service: switch.turn_on
          data:
            entity_id: switch.radiateur_soan
        icon_template: "{% if is_state('switch.radiateur_soan', 'on') %}mdi:radiator-disabled{% else %}mdi:radiator{% endif %}"
```

Adapte les actions turn_on / turn_off a ton cas et ça devrait marcher. et tu mets ce switch en entrée du VTherm (over_switch donc).

Le mode over_climate est beaucoup plus limité et devrait être réservé au cas impossible à faire autrement. Avec le mode over_switch tu vas avoir nativement à toutes les fonctions du VTherm (régulation, fenêtre, mouvement, gestion de la puissance, mode sécurté, ...)

**Reply by SchottyOne** [reply]
Bonjour Jean-Marc et merci beaucoup pour la réactivité. C'est appréciable.
Il s'avère qu'en postant ma demande, au fur et à mesure d'écrire je me suis persuadé que la solution pertinente tournait autour de cette notion de switch. 
Nativement, je n'ai pas trouvé de notion de switch dans les attributs du boitier (vu comme entité climate)
![image](https://github.com/jmcollin78/versatile_thermostat/assets/124760807/9f22024d-4796-4ff6-9995-2579902727be)
La ou je pensais que ça allait fonctionner est que le VTHERM semble reprendre tous les attributs du boitier et que le fait de sélectionner les preset (eco, confort, boost) allait simplement matcher avec les modes du boitier (off pour eco, heat pour confort et boost par exemple) Mais cela n'a aucune action. les attributs du VTHERM
![image](https://github.com/jmcollin78/versatile_thermostat/assets/124760807/57123ede-15a3-4e4f-abcd-8d189f53ce0a)
En revanche, les modes manuels sont bien repris sur la card et fonctionnent.
![image](https://github.com/jmcollin78/versatile_thermostat/assets/124760807/56cad97a-56d9-4ae0-bd8a-0b9d7f268c61)
Mais du coup, avant que tu répondes, j'ai en effet fait des tests en créant un switch quelconque dans HA et en effet, VTHERM associé aux sondes, aux capteurs d'ouverture...Semble bien piloter l'état du switch.
Du coup, je vais partir sur ta proposition en sacrifiant les modes du boitier aux seuls modes on/off (Heat, off du coup) les modes eco, confort..qui seront du coup gérés par VTHERM. Je vais suivre les exemples que tu me fournis car je ne suis pas un pro du code ni de HA mais je devrais m'en sortir ;) 

J'aurais une question quand tu indiques de mettre confort à 24°, tu sous-entend de régler cette température sur le convecteur lui-même ? Pour qu'il évite de couper trop tôt par exemple ? ou de régler cette température sur les preset confort du VTHERM ? Sachant que mon objectif est d'être sur une consigne à 21° en mode confort.

Je me lance dans cette configuration et n'hésiterait pas à te tenir au courant des avancées. J'ai failli finir par partir sur automatisations dans tous les sens pour couper les convecteurs une fois que les bonnes températures étaient atteintes...

Merci encore pour cette aide précieuse.

Pascal

**Reply by jmcollin78** [reply]
> J'aurais une question quand tu indiques de mettre confort à 24°, tu sous-entend de régler cette température sur le convecteur lui-même ? Pour qu'il évite de couper trop tôt par exemple ? 

Oui c'est ça. C'est à VTherm de couper et pas à ton convecteur. Mais faut pas non plus mettre trop haut, ca sert de sécurité. La conseil c'est de mettre la temp max que tu auras un jour + 2°.

**Reply by SchottyOne** [reply]
OK, bien noté. Je me lance..

**Comment by SchottyOne** [comment]
Bonjour Jean-Marc, je me permets de te faire un update suite à mes premiers essais.
Tout d'abord, il semble que le thermostat remplisse bien son rôle en passant par le "Over_switch" suite à la création des switchs en suivant le template que tu m'as fourni 😃 Ce qui était le but recherché. J'ai passé les consignes des radiateurs à 23° pour une consigne Vtherm à 21° (idem +2° pour chaque mode..)
Je reste sous surveillance pour vérifier le comportement car en fait mon souci n'est plus tant sur le pilotage des radiateurs par le vtherm (qui semble fonctionner) que dans l'analyse de son comportement.
En effet, j'imagine que soit je dois revoir l'adaptation du template ou que je doive chercher un peu dans HA car les états du switch ne sont pas pertinents (mais je dirais que cela sort de ton périmètre pour le coup)
Le template que j'ai appliqué sachant que mes boitiers sont sous vu sous HA comme entité type climate.chauffage..) Et qu'ils font donc appel aux service "set_hvac_mode.." pour le on/off (Heat/Off)
```
- platform: template
  switches:
    chauffage_chambre:
      friendly_name: switch_chauffage_chambre
      value_template: "{{ is_state('switch.chauffage_chambre', 'off') }}"
      turn_on:
        service: climate.set_hvac_mode
        data:
          hvac_mode: heat
        target:
          entity_id: climate.chauffage_chambre
      turn_off:
        service: climate.set_hvac_mode
        data:
          hvac_mode: "off"
        target:
          entity_id: climate.chauffage_chambre
      icon_template: "{% if is_state('switch.chauffage_chambre', 'off') %}mdi:radiator-disabled{% else %}mdi:radiator{% endif %}"
```
La commande manuelle en allant sur l'entité switch créée fonctionne bien (ca allume et éteint le radiateur)
En revanche le statut de l'entité (et l'icone) reste toujours dans l'état inactif (désactivé) peut importe si il est on ou off. 
![image](https://github.com/jmcollin78/versatile_thermostat/assets/124760807/33156689-342b-4c6e-b390-4e29105be15f)
![image](https://github.com/jmcollin78/versatile_thermostat/assets/124760807/0f2a5d55-fdd2-42c3-b69e-3e9ea524d2ab)

Sur une card entité, le statut de l'entité reste off et si con clique pour la basculer en ON, l'interrupteur revient à off.
![2023-11-22_13h22_16](https://github.com/jmcollin78/versatile_thermostat/assets/124760807/d2f2d336-7347-4c76-80b3-3bb1c7e031a8)

Je crois avoir déjà parcouru un article sur ce type de comportement donc je vais cherché de mon côté.

En tout ca comme indiqué au départ, ça semble faire le job 😉 

J'aurai une question au passage (mais je vais vérifier dans le readme si besoin) Concernant les entités liées à l'énergie.
Actuellement, je n'ai pas activé l'option "gestion de la puissance" 
Néanmoins, je vois des données apparaitre sur certaines attributs du VTHERM et j'avoue ne pas comprendre d'ou il récupère ces valeurs.
![image](https://github.com/jmcollin78/versatile_thermostat/assets/124760807/2e6d0d97-4023-4509-aa8b-e5a4f28d34ce)
J'ai bien des sensors dans mon installation pour les mesures électriques (Zlinky, sensors Emporia VUE sur mes départ élec) Mais je ne vois pas de lien avec VTHERM qui pourrait faire le match entre ces données.
Rien de bien grave, juste de la curiosité.

En tout cas, merci encore de m'avoir aidé et mis sur la bonne piste.

Pascal

**Reply by jmcollin78** [reply]
Si l'état revient tout seul c'est que l'état de ton template ne change pas. Donc faut regarder ça : `value_template: "{{ is_state('switch.chauffage_chambre', 'off') }}"`
Si ce que je comprends c'est qu'il s'auto-reference, ca ne va pas marcher.

L'énergie provinet forcément d'une puissance d'équipement qui a été saisie dans la config du VTherm. Ca ne peut pas être autrement. Peut-être que la case est décochée mais si tu as saisies les valeurs un jour, elles sont peut être toujours actives (mais tu ne les vois plus car la case est décochée). Je crois que j'ai une ano comme ça.

**Reply by jmcollin78** [reply]
Et pour toutes les questions hors VTHerm strictement je conseile https://forum.hacf.fr (en FR donc)

**Comment by SchottyOne** [comment]
En effet, je cherche des pistes pour corriger mon template. J'en ai peut être trouvé une mais comme je ne maitrise pas les syntaxes, je vais faire des tests (usage is_state_attr au lieu de is_state) 
https://community.home-assistant.io/t/make-my-own-climate-entity-with-a-temp-sensor-and-a-climate-entity/510458/10
Concernant l'énergie, effectivement lors de mes premières config j'avais validé l'option mais j'avais buté car je n'avais pas trouvé d'entité qui permette de récupérer la puissance max. Cela doit trainer dans la config.
Je continue de creuser ;)

**Reply by jmcollin78** [reply]
Le forum hacf.fr est parfait pour ça

**Comment by SchottyOne** [comment]
Bon en fait ton coup de 'auto référencement' du switch m'a mis sur la bonne piste je pense
J'ai corrigé 
`value_template: "{{ is_state('switch.chauffage_chambre', 'off') }}"` par 
`value_template: "{{ is_state('climate.chauffage_chambre', 'heat) }}" `
Pour vérifier l'état du boitier et avoir la bonne visualisation de l'état du switch quand il est à ON.
Ca a l'air de matcher déjà sur le contrôle du boitier par l'action manuelle sur le switch.
![2023-11-25_10h44_38](https://github.com/jmcollin78/versatile_thermostat/assets/124760807/b03ab848-b30e-4249-b754-9b2e08d1fc03)
![image](https://github.com/jmcollin78/versatile_thermostat/assets/124760807/c5918a5d-d396-483c-afb6-f1073d45550e)
Je vais laisser tourner comme ça pour voir ce que cela donne sur VTHERM.
